frontend/src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import {
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Button,
+  NumberInput,
+  NumberInputField,
+  IconButton,
+  Flex,
+  Box,
+  Text,
+  GridItem,
+} from "@chakra-ui/react";
+
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronDownIcon,
+  TriangleDownIcon,
+} from "@chakra-ui/icons";
+
+type Props = {
+  numRecords: number;
+  resultsPerPage: number;
+  pageNum: number;
+  setResultsPerPage: React.Dispatch<React.SetStateAction<number>>;
+  getLogRecords: (page_number: number) => Promise<void>;
+};
+
+const RESULTS_PER_PAGE_OPTIONS = [10, 25, 50];
+
+const Pagination = ({
+  numRecords,
+  resultsPerPage,
+  pageNum,
+  setResultsPerPage,
+  getLogRecords,
+}: Props): React.ReactElement => {
+  const numPages = Math.ceil(numRecords / resultsPerPage);
+
+  const handleNumberInputChange = (
+    newPageNumString: string,
+    newPageNum: number,
+  ) => {
+    if (newPageNum !== pageNum && newPageNum >= 1 && newPageNum <= numPages) {
+      getLogRecords(newPageNum);
+    }
+  };
+
+  const getNumRecordsStr = () => {
+    return `${numRecords} ${numRecords === 1 ? "item" : "items"}`;
+  };
+
+  return (
+    <Box padding="12px 0px 33px">
+      <Flex justifyContent="space-between" alignItems="center">
+        <Box>
+          <Text color="#6D8788">{getNumRecordsStr()}</Text>
+        </Box>
+        <Box>
+          <Flex alignItems="center" justifyContent="space-evenly">
+            <IconButton
+              className="icon-button"
+              variant="ghost"
+              aria-label="Previous page"
+              icon={<ChevronLeftIcon boxSize={7} color="#6D8788" />}
+              disabled={pageNum <= 1}
+              onClick={() => (pageNum > 1 ? getLogRecords(pageNum - 1) : null)}
+            />
+            <Flex
+              alignItems="center"
+              justifyContent="space-evenly"
+              padding="0px 25px 0px"
+              width="210px"
+            >
+              <Text>Page</Text>
+              <NumberInput
+                maxW="60px"
+                value={pageNum}
+                size="sm"
+                onChange={handleNumberInputChange}
```

Comment:
not a big fan of how the user can't press backspace on this input... here's my suggestion of how it should behave:
- user can backspace the number in the input. No number should display in the input, however, the log records will remain the same (it'll still be on the same page)
- When the user types in a new number, it'll switch to that page
- If the user clicks away without inputting a new page number, it'll reset to the original page